rust/altrios-core/src/train/speed_limit_train_sim.rs

```diff
@@ -288,31 +290,44 @@ impl SpeedLimitTrainSim {
     }
 
     pub fn solve_step(&mut self) -> anyhow::Result<()> {
+
+        //set catenary power limit.  This was the same as set_speed_train_sim.  I am assuming it is 0 right now.
         self.loco_con
             .set_cat_power_limit(&self.path_tpc, self.state.offset);
+        //set aux power for the consist.  Very similar to set speed train sim.  boolean argument is whether engine is on.  Not a huge fane of this argument.  I think this should be baked into a loco control system in locomotive.rs
         self.loco_con.set_pwr_aux(Some(true))?;
+        //set the maximum power out based on dt.  This is different than set speed train sim because that one calls the speed trace dt.
+        //why are we not passing in aux power here and self.state?  Am I missing something obviouls here?
```

Comment:
@SWRIganderson, as we have it now, aux load is determined entirely at the locomotive level and is solely dependent on a constant and tractive-power coefficient.  As such, there's no need to pass this in.  This obviously should be more sophisticated to account for other factors that impact aux load.   I have lots of thoughts, but they all need vetting from another engineer, e.g. you!

Reply:
I agree that we should have something more sophisticated.  I think this needs to go down to the component/subsystem level so that we can capture differences between engine, battery, or traction motor aux loads.  Not sure how we will validate against data though.

Reply:
@SWRIganderson , we've done some of this type of stuff in FASTSim.   I am 98% sure I have multiple issues saying that we need better aux modeling so we'll get back to it when it becomes important.